server/schedule/operator.go

```diff
@@ -181,8 +183,10 @@ func (o *Operator) Step(i int) OperatorStep {
 func (o *Operator) Check(region *core.RegionInfo) OperatorStep {
 	for step := atomic.LoadInt32(&o.currentStep); int(step) < len(o.steps); step++ {
 		if o.steps[int(step)].IsFinish(region) {
+			operatorStepDuration.WithLabelValues(reflect.TypeOf(o.steps[int(step)]).Name()).Observe(time.Since(o.stepTime).Seconds())
 			atomic.StoreInt32(&o.currentStep, step+1)
 		} else {
+			o.stepTime = time.Now()
```

Comment:
if two times check when step not finished, it will update the start time.

Reply:
And the `stepTime` is not set up when creating the operator.

Reply:
@nolouch that is for purpose, we don't need care about the first check that step not finished finally. For the second check, it updates the start time, and then we can get the correct elapsed time of the second try.